gojmx/types.go

```diff
@@ -57,6 +62,38 @@ func (j *AttributeResponse) GetValue() interface{} {
 	}
 }
 
+// GetValueAsFloat casts the value from AttributeResponse to float based on type.
+func (j *AttributeResponse) GetValueAsFloat() (float64, error) {
+	switch (*j).ResponseType {
+	case ResponseTypeBool:
+		if j.BoolValue {
+			return 1, nil
+		}
+		return 0, nil
+	case ResponseTypeString:
+		parsedValue, err := strconv.ParseFloat(fmt.Sprintf("%v", j.StringValue), 64)
+		if err != nil {
+			return 0, err
+		}
+		if isNaNOrInf(parsedValue) {
+			return 0, ErrNonNumeric
+		}
+		return parsedValue, nil
+	case ResponseTypeDouble:
+		return j.DoubleValue, nil
+	case ResponseTypeInt:
+		return float64(j.IntValue), nil
+	case ResponseTypeErr:
+		return 0, nil
+	default:
+		panic(fmt.Sprintf("unkown value type: %v", j.ResponseType))
```

Comment:
Can we return an error here, instead of panicking?

Reply:
This is how it's done in the other GetValue, I am replying the same behaviour.

Reply:
Is a valid case of panicking, the types are defined in the protocol and should only panic when the java module and go module doesn't respect the procol

Reply:
noted